### ISSUE TITLE
BulkAdd: keys argument invalid on tables with inbound keys

### ISSUE THREAD

**Issue by HurricaneInteractive** [dfahlander/Dexie.js#1047]
Currently trying to use the [bulkAdd](https://dexie.org/docs/Table/Table.bulkAdd()) method, however, I am getting the following error.

```
Unhandled Rejection (InvalidArgumentError): bulkAdd(): keys argument invalid on tables with inbound keys
```

### Code

I am using Typescript and I have defined a database that extends `Dexie` and the table structure is as follows.

```ts
this.version(1.0).stores({
  items: "++id, [container+client], client, completed, description, reviewBy"
})
```

Then when calling the `bulkAdd` method I have the following function.
```ts
class UseCase {
  // ... removed
  
  public bulkAdd(items: D[]): Promise<K[]> {
    return this.database.transaction("rw", this.table.name, () => {
      return this.table.bulkAdd<true>(items, { allKeys: true })
      // I've also tried
      // return this.table.bulkAdd<true>(items, [], { allKeys: true })
      // return this.table.bulkAdd(items, { allKeys: true })
    })
  }
}
```

Finally, within a method in a React component, I do the following:

```tsx
const Component = () => {
  const submit = async () => {
    const items: D[] = [
      { container: "", client: 1, completed: false, description: "", reviewBy: new Date() },
      { container: "", client: 1, completed: false, description: "", reviewBy: new Date() },
      { container: "", client: 1, completed: false, description: "", reviewBy: new Date() },
      { container: "", client: 1, completed: false, description: "", reviewBy: new Date() },
      { container: "", client: 1, completed: false, description: "", reviewBy: new Date() }
    ]

    const ids = await UseCaseImplementation.bulkAdd(items)
  }

  return ...
}
```

From what I understand `++id` is a "inbound" key so I should not be passing that along to the call, which I am not doing. I also looked at the definitions for the `bulkAdd` method and I can see the following. Which makes it seem like I can call the function and omit the `keys` parameter.

```ts
bulkAdd<B extends boolean>(items: T[], options: { allKeys: B }): PromiseExtended<B extends true ? TKey[] : TKey>;
```

However, I think, it is calling this function instead and thinks that the options I am passing are keys.

```ts
bulkAdd<B extends boolean>(items: T[], keys: IndexableTypeArrayReadonly, options: { allKeys: B }): PromiseExtended<B extends true ? TKey[] : TKey>;
```

### Versions
- Dexie: 3.0.1
- Typescript: 3.7.2

**Notes:** This is for a client so no repo is available to share.

**Comment by dfahlander** [comment]
I've reviewed the code and the only logical reason this happens is the following:

1. The app has in the past distributed the same database and version number (1) in which the table had outbound keys.
2. This user loaded the app at that point.
3. A new version of the app was published where the table was changed to being inbound, but the version number of the database was not incremented.

Could that be the case? If so, the solution is:

* Increase the version number to 2 and publish the app again. Tell the user to try again.
